### DISCUSSION TITLE
Setup mkdocs-materials

### DISCUSSION THREAD

**Discussion by gh-repo-status-check** [squidfunk/mkdocs-material#7793] [Q&A] [answered]
How can I switch from urls like docs.example.com/setup/docker.html to docs.example.com/setup/docker. How does that work?

**Comment by niclasheinz** [comment]
Can you please add some context to the setup? For example, which plugins are activated?

**Reply by gh-repo-status-check** [reply]
I use the following plugins:

- privacy
- offline
- search
- git-revision-date-localized
- minify
- social
- tags
- mkdocs-gallery
- print-site

**Comment by kamilkrzyskow** [comment, marked as answer]
Hello @gh-repo-status-check,
the way links are processed is related to the [use_directory_urls](https://www.mkdocs.org/user-guide/configuration/#use_directory_urls) config option. MkDocs generates static HTML pages, and it's either `.../docker.html` or `.../docker/index.html`, where the latter can be opened with `.../docker/`. Notice the `/` at the end which "opens" the directory, and the web browser automatically loads the `index.html` file.

There is no official way to setup direct link to `docker.html` pages without the `.html` part. You could theoretically modify the links with a [hook](https://www.mkdocs.org/user-guide/configuration/#hooks), however some plugins could rely on the `.html` suffix for processing, so I wouldn't personally try to hack it. Therefore my recommendation is to enable the directory urls (which is the default), and use the links with the trailing slash.

Similar discussions:
- https://github.com/squidfunk/mkdocs-material/discussions/6332
- https://github.com/squidfunk/mkdocs-material/discussions/6046
- I recall there was one more, which I had in mind, but can't find it either here or in the other discussion board 🤔

**Reply by gh-repo-status-check** [reply]
Thanks for your reply @kamilkrzyskow this setting is enabled, but I have still these .html urls

**Reply by kamilkrzyskow** [reply]
Then please create a minimal runnable reproduction archive, following this [guide](https://squidfunk.github.io/mkdocs-material/guides/creating-a-reproduction/), which will help to pinpoint the issue, to either a plugin / misconfiguration. 

Perhaps a plugin like, the `offline` plugin disables the directory links during the processing, as stated in https://github.com/squidfunk/mkdocs-material/discussions/6046 it should be used only for local copy generation, not remote server.
https://github.com/squidfunk/mkdocs-material/blob/7fb3a39bbc17865743b8cb8846aa4e845ac53eb4/material/plugins/offline/plugin.py#L41

Make use of the `!ENV` syntax to conditionally enable such plugins only when needed:
- https://squidfunk.github.io/mkdocs-material/plugins/offline/#config.enabled

**Reply by gh-repo-status-check** [reply]
thanks for your time @kamilkrzyskow it works now.

**Reply by niclasheinz** [reply]
I have created a PR #7796 so that users can easily see this restriction in the official documentation
